lib/mpl_toolkits/tests/test_mplot3d.py

```diff
@@ -491,3 +491,18 @@ def test_autoscale():
     ax.set_autoscalez_on(True)
     ax.plot([0, 2], [0, 2], [0, 2])
     assert ax.get_w_lims() == (0, 1, -.1, 1.1, -.4, 2.4)
+
+
+@pytest.mark.parametrize('value', [np.inf, np.nan])
+@pytest.mark.parametrize(('setter', 'side'), [
+    (plt.figure().add_subplot(111, projection='3d').set_xlim3d, 'left'),
```

Comment:
Cleanup will work on the test, but I don't think it will work right on the parametrization. Consequently, I don't think it makes sense to create the figure in this list. It should be created in the test itself and only the setter name should be here. You can use `getattr` to dynamically access the method you want.